torch/utils/_cxx_pytree.py

```diff
@@ -660,6 +660,17 @@ def tree_map_only(
     ...
 
 
+@overload
+def tree_map_only(
+    type_or_types_or_pred: TypeAny,
+    /,
+    func: FnAny[Any],
+    tree: PyTree,
+    is_leaf: Optional[Callable[[PyTree], bool]] = None,
```

Comment:
A lot of the types for PyTree etc in this file are Any, which doesn't actually help type checking much. We really should replace a lot of them with TypeVar...